test/test_cuda.py

```diff
@@ -425,6 +425,79 @@ def tmp(self):
 
 
 class TestCuda(TestCase):
+
+    def test_memory_allocated_stats(self):
+        torch.cuda.empty_cache()
+        m0 = last_m = torch.cuda.memory_allocated()
+        max_m = torch.cuda.max_memory_allocated()
+
+        def assert_change(comp=1):
+            # comp > 0: increased
+            # comp = 0: equal
+            # comp < 0: decreased
+            nonlocal last_m, max_m
```

Comment:
`nonlocal` is Python 3 only. You can do `last_m = [torch.cuda.memory_allocated()]` and then use `last_m[0]` everywhere